src/zenml/post_execution/pipeline_run.py

```diff
@@ -169,11 +177,34 @@ def get_step(self, name: str) -> StepView:
             KeyError: If there is no step with the given name.
         """
         self._ensure_steps_fetched()
+        if step and step_name:
+            raise TypeError(
+                "'step' and 'step_name' both set for the"
+                "get_step() method. This is not supported. "
+                "Please set either the 'step' or the "
+                "'step_name' parameter."
+            )
+        elif step:
+            if isinstance(step, zenml.steps.base_step.BaseStepMeta):
+                step_name = step.__name__
+            elif isinstance(step, zenml.steps.base_step.BaseStep):
+                step_name = step.name
+        elif "name" in kwargs and not step_name:
+            logger.warning(
+                "Using 'name' to get a step from "
```

Comment:
super nit. Warnings like this can be much more friendly for newcomers if you can put a link to the specific API docs containing examples. `https://apidocs.zenml.io/latest/api_docs/post_execution/#zenml.post_execution.pipeline_run.PipelineRunView.get_step`


you can streamline this as

```
def api_link() -> str:
   # stack trace get caller name
   return "https://apidocs.zenml.io/latest/api_docs/post_execution/#{caller_name}"
```

Reply:
I love this suggestion, I'll see what I can do within this ticket to make the warnings more user friendly.